python/pyarrow/tests/test_fs.py

```diff
@@ -240,27 +488,26 @@ def identity(v):
         ('gzip', 256, gzip_compress),
     ]
 )
-def test_open_input_stream(fs, tempdir, testpath, compression, buffer_size,
-                           compressor):
-    file = testpath('abc')
-    file_ = tempdir / 'abc'
-    data = b'some data' * 1024
-    file_.write_bytes(compressor(data))
+def test_open_input_stream(fs, compression, buffer_size, compressor):
+    _p, p = fs.pathpair('open-input-stream')
 
-    with fs.open_input_stream(file, compression, buffer_size) as f:
-        result = f.read()
+    data = b'some data for reading' * 1024
+    fs.write_bytes(_p, compressor(data))
+
+    with fs.impl.open_input_stream(p, compression, buffer_size) as f:
+        result = f.read(len(data))
```

Comment:
In case of S3 `f.read()` fails because of "over-reading":

```
>   raise ArrowIOError(message)
E   pyarrow.lib.ArrowIOError: AWS Error [code 100]: Unable to parse ExceptionName: InvalidRange Message: The requested range is not satisfiable with address : 127.0.0.1
E   In ../src/arrow/filesystem/s3fs.cc, line 345, code: GetObjectRange(client_, path_, position, nbytes, &result)
E   In ../src/arrow/filesystem/s3fs.cc, line 375, code: ReadAt(pos_, nbytes, bytes_read, out)
E   In ../src/arrow/io/buffered.cc, line 395, code: raw_->Read(bytes_to_read, bytes_read, reinterpret_cast<uint8_t*>(out) + bytes_buffered_)
E   In ../src/arrow/io/buffered.cc, line 415, code: Read(nbytes, &bytes_read, buffer->mutable_data())
E   In ../src/arrow/io/compressed.cc, line 293, code: raw_->Read(kChunkSize, &compressed_)
E   In ../src/arrow/io/compressed.cc, line 364, code: EnsureCompressedData()
E   In ../src/arrow/io/compressed.cc, line 394, code: RefillDecompressed(&decompressor_has_data)
```

@pitrou should we fix it in this PR or in a follow-up?

Reply:
Can fix in a follow-up. Can you open a JIRA for it?

Reply:
Resolved it.